protocol/contracts/libraries/Silo/LibSilo.sol

```diff
@@ -547,34 +553,37 @@ library LibSilo {
     /**
      * @dev returns the amount of `plenty` an account has.
      */
-    function balanceOfPlenty(address account) internal view returns (uint256 plenty) {
+    function balanceOfPlenty(address account, address well) internal view returns (uint256 plenty) {
         AppStorage storage s = LibAppStorage.diamondStorage();
         Account.State storage a = s.a[account];
-        plenty = a.sop.plenty;
+        plenty = a.sop[well].plenty;
         uint256 previousPPR;
 
         // If lastRain > 0, then check if SOP occured during the rain period.
         if (s.a[account].lastRain > 0) {
             // if the last processed SOP = the lastRain processed season,
             // then we use the stored roots to get the delta.
-            if (a.lastSop == a.lastRain) previousPPR = a.sop.plentyPerRoot;
-            else previousPPR = s.sops[a.lastSop];
-            uint256 lastRainPPR = s.sops[s.a[account].lastRain];
+            if (a.lastSop == a.lastRain) {
+                previousPPR = a.sop[well].plentyPerRoot;
+            } else {
+                previousPPR = s.sops[a.lastSop][well];
```

Comment:
appreciate you expanding the if else block

Reply:
🫡 prefer this way as well, easier to read.